app/cdap/components/GridTable/index.tsx

```diff
@@ -0,0 +1,286 @@
+      nullValue = nonNullValue.null ? (nonNullValue.null / 100) * lengthOfData : 0;
+      empty = nonNullValue.empty ? (nonNullValue.empty / 100) * lengthOfData : 0;
+      count = parseInt(nullValue + empty);
+    }
+    return count;
+  };
+
+  const checkFrequentlyOccuredValues = (key) => {
+    const valueOfKey = gridData.values.map((el) => el[key]);
+    let mostfrequentItem = 1;
+    let count = 0;
+    let item = '';
+    const data = {
+      name: '',
+      count: 0,
+    };
+    for (let i = 0; i < valueOfKey.length; i++) {
+      for (let j = i; j < valueOfKey.length; j++) {
+        if (valueOfKey[i] == valueOfKey[j]) {
+          count++;
+        }
+        if (mostfrequentItem < count) {
+          mostfrequentItem = count;
+          item = valueOfKey[i];
+        }
+      }
+      count = 0;
+      item = item == '' ? valueOfKey[i] : item;
+    }
+    data.name = item;
+    data.count = mostfrequentItem;
+    return data;
+  };
+
+  const createMissingData = (statistics) => {
+    const objectArray = Object.entries(statistics);
+    const metricArray = [];
+    objectArray.forEach(([key, value]) => {
+      const valueToArray = Object.entries(value);
+      const tempArray = [];
+      valueToArray.forEach(([vKey, vValue]) => {
+        tempArray.push({
+          label:
+            vKey == 'general' && convertNonNullPercent(key, vValue) == 0
+              ? checkFrequentlyOccuredValues(key).name
+              : vKey == 'general'
+              ? 'Missing/Null'
+              : vKey == 'types'
+              ? ''
+              : '',
+          count:
+            vKey == 'types'
+              ? ''
+              : convertNonNullPercent(key, vValue) == 0
+              ? checkFrequentlyOccuredValues(key).count
+              : convertNonNullPercent(key, vValue),
+        });
+      }),
+        metricArray.push({
+          name: key,
+          values: tempArray.concat(invalidCountArray),
+        });
+    });
+    return metricArray;
+  };
+
+  const getGridTableData = async () => {
+    const rawData: any = gridData;
+    const headersData = createHeadersData(rawData.headers, rawData.headers, rawData.types);
+    setHeadersNamesList(headersData);
+    if (rawData && rawData.summary && rawData.summary.statistics) {
+      const missingData = createMissingData(gridData.summary.statistics);
+      setMissingDataList(missingData);
+    }
+    const rowData =
+      rawData &&
+      rawData.values &&
+      Array.isArray(rawData.values) &&
+      rawData.values.map((eachRow) => {
+        const { body, ...rest } = eachRow;
+        return rest;
+      });
+
+    setRowsDataList(rowData);
+  };
+
+  useEffect(() => {
+    getGridTableData();
+  }, [gridData]);
+
+  return (
+    <Box className={classes.wrapper}>
+      <BreadCrumb datasetName={wid} />
+      <Table aria-label="simple table" className="test">
+        <TableHead>
+          <TableRow>
+            {Array.isArray(headersNamesList) &&
```

Comment:
is there any way this won't be an array? you're initializing it as [] do you set it to be something else?

Reply:
Done